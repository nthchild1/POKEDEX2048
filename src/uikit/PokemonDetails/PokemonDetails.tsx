import React from 'react';
import {Text, View} from 'react-native';
import {normalizePx} from '../../utils/utilFunctions';
import * as Progress from 'react-native-progress';
import {colors} from '../../constants';
import styled from 'styled-components/native';

interface PokemonDetailsProps {
  isExpanded: boolean;
  pokemon: object;
  pokemonType: string;
}

const Container = styled.View`
  background-color: ${({pokemonRed}) => pokemonRed};
  flex: 1;
  margin-right: ${normalizePx(20)}px;
  margin-top: ${normalizePx(35)}px;
  margin-bottom: ${normalizePx(35)}px;
  border-radius: 15px;
  justify-content: center;
  align-items: flex-start;
  overflow: hidden;
`;

const Label = styled.Text`
  color: black;
  font-size: 10px;
  font-weight: bold;
`;

const Value = styled.Text`
  color: white;
`;

export const Stats = styled.View`
  background-color: white;
  flex: 1;
  width: 100%;
  padding: 15px;
  align-items: center;
`;

function PokemonDetails({
  pokemon,
  pokemonType,
  isExpanded,
}: PokemonDetailsProps): JSX.Element {
  const {pokemonRed} = colors;

  return isExpanded ? (
    <Container {...{pokemonRed}}>
      <View style={{padding: '5%'}}>
        <Label>
          HEIGHT : <Value>{pokemon.height} ft</Value>
        </Label>
        <Label>
          TYPE : <Value> {pokemonType}</Value>
        </Label>
        <Label>
          ABILITIES :
          <Value>
            {pokemon?.abilities?.map(({ability}) => ability.name).join(', ')}
          </Value>
        </Label>
      </View>
      <Stats>
        {pokemon?.stats?.map(stat => {
          return (
            <View key={stat?.stat?.name + ' : ' + stat?.base_stat}>
              <Text style={{color: 'black', fontSize: 10}}>
                {stat?.stat?.name + ' : ' + stat?.base_stat}
              </Text>
              <Progress.Bar progress={stat?.base_stat / 100} width={100} />
            </View>
          );
        })}
      </Stats>
    </Container>
  ) : (
    (null as unknown as JSX.Element)
  );
}

PokemonDetails.defaultProps = {
  isExpanded: true,
  pokemonType: 'fire',
  pokemon: {
    height: 5,
    id: 5,
    sprites: {
      front_default: '',
    },
    types: {
      type: {
        name: 'fire',
      },
    },
    abilities: [
      {
        ability: {
          name: 'overgrow',
        },
      },
    ],
    stats: [
      {
        stat: {
          name: 'hp',
        },
        base_stat: 54,
      },
    ],
  },
};

export default PokemonDetails;
