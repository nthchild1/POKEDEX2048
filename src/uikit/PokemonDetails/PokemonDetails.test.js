import React from 'react';
import PokemonDetails, {Stats} from './PokemonDetails';
import renderer from 'react-test-renderer';

const mockPockemon = {
  height: 5,
  id: 5,
  sprites: {
    front_default: '',
  },
  types: {
    type: {
      name: 'fire',
    },
  },
  abilities: [
    {
      ability: {
        name: 'overgrow',
      },
    },
  ],
  stats: [
    {
      stat: {
        name: 'hp',
      },
      base_stat: 54,
    },
  ],
};

it('renders correctly', () => {
  const tree = renderer.create(
    <PokemonDetails pokemon={mockPockemon} pokemonType={'fire'} />,
  );
  expect(tree).toMatchSnapshot();
});

it('should not render if isExpanded is false', () => {
  const testRenderer = renderer.create(
    <PokemonDetails
      isExpanded={false}
      pockemon={mockPockemon}
      pockemonType={'fire'}
    />,
  );
  const testInstance = testRenderer.root;
  expect(testInstance.children.length).toBe(0);
});

it('should render a list of bars showing abilities', function () {
  const testRenderer = renderer.create(
    <PokemonDetails
      isExpanded={true}
      pockemon={mockPockemon}
      pockemonType={'fire'}
    />,
  );
  const testInstance = testRenderer.root;

  expect(testInstance.findByType(Stats).children.length).toBeGreaterThan(0);
});
