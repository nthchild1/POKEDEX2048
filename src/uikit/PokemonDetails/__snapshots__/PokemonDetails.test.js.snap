// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  pokemonRed="#dd4e4e"
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#dd4e4e",
        "borderRadius": 15,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginBottom": 35,
        "marginRight": 20,
        "marginTop": 35,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Object {
        "padding": "5%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      HEIGHT : 
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
          ]
        }
      >
        5
         ft
      </Text>
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      TYPE : 
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
          ]
        }
      >
         
        fire
      </Text>
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      ABILITIES :
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
          ]
        }
      >
        overgrow
      </Text>
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 10,
          }
        }
      >
        hp : 54
      </Text>
      <View
        animated={true}
        animationConfig={
          Object {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={false}
        indeterminateAnimationDuration={1000}
        onLayout={[Function]}
        progress={0.54}
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderColor": "rgba(0, 122, 255, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "overflow": "hidden",
              "width": 100,
            },
            undefined,
          ]
        }
        useNativeDriver={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 122, 255, 1)",
              "height": 6,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateX": -22.54,
                },
                Object {
                  "scaleX": 0.540046,
                },
              ],
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
